controllers/sriovoperatorconfig_controller.go

```diff
@@ -147,16 +147,48 @@ func (r *SriovOperatorConfigReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
+// This function will attempt to render the plugin daemon sets to retrieve the default node selector
+// when successful it will set the input node selector.
+func (r *SriovOperatorConfigReconciler) getDefaultPluginDsNodeSelector(name string, m *map[string]string) error {
```

Comment:
I think a better solution is just to have the default nodeSelector for the device plugin as a variable in the operator and just render it

Reply:
Do you mean that in the template, we should hardcode the default nodeSelector? If yes, I agree. That would be concise.

Reply:
I mean to have the nodeSelector in the API helper as a map
```
nodeSelector:
        kubernetes.io/os: linux
        node-role.kubernetes.io/worker:
```
and in the yaml we have 

```
nodeSelector:
  {nodeSelector}
```

so we don't need to parse the yaml just for that

Reply:
@SchSeba Thanks for the suggestion. I did think about that but in Openshift, we have "beta.kubernetes.io/os=linux", but in upstream we have "kubernetes.io/os=linux". Thus it doesn't match correctly in OCP.

Not sure at which point OCP coverts kubernetes.io/os=linux => beta.kubernetes.io/os=linux

Reply:
^ Sorry scratch this comment. Trying a different method now.

Reply:
@SchSeba Please see the new solution now. I think it is a lot cleaner.